### PR TITLE
Theus: prometheus logging across bittensor components

### PR DESCRIPTION
**Pull request by unconst**
Prometheus logging:

Bittensor and miner classes can enable a local Prometheus server that logs statistics. These can be picked up by a global Prometheus server for use in network monitoring.

Features:

Global Prometheus switch
--prometheus.level <DEBUG, INFO, OFF>

Object level Prometheus switches:
--config.promehteus.level <DEBUG, INFO, OFF>
--axon.promehteus.level <DEBUG, INFO, OFF>
--dendrite.promehteus.level <DEBUG, INFO, OFF>

Turning off the global Prometheus switch ensures the main Prometheus server is not created. 

Additional changes:
-- Removes defunct wandb statistics logging from Axon and Dendrite and from Validator and Miner classes.

### CODE REVIEW

File: bittensor/_neuron/text/core_validator/__init__.py [hunk 1]
```diff
@@ -137,14 +142,26 @@ def __init__(
...
+            port = config.prometheus.port if config.axon.port == bittensor.defaults.axon.port else config.axon.port - 1000
+        )
+
         # === Create Bittensor objects ===
-        bittensor.logging( config = self.config, logging_dir = self.config.neuron.full_path )
         self.wallet = bittensor.wallet ( config = self.config ) if wallet == None else wallet
         self.subtensor = bittensor.subtensor ( config = self.config ) if subtensor == None else subtensor
         self.metagraph = bittensor.metagraph ( config = self.config, subtensor = self.subtensor ) if metagraph == None else metagraph
         self.dendrite = bittensor.dendrite ( config = self.config, wallet = self.wallet ) if dendrite == None else dendrite
+        self.axon = bittensor.axon ( config = self.config, wallet = self.wallet ) if axon == None else axon
```
Comment (Eugene-hu) [inline bittensor/_neuron/text/core_validator/__init__.py:164]: Why do we need the axon for the validator? Is it for the Prometheus server?

**Review by Eugene-hu — APPROVED** [review]
LGTM; left you a comment

File: bittensor/_neuron/text/core_validator/__init__.py (hunk 1 above)
Comment (unconst) [reply bittensor/_neuron/text/core_validator/__init__.py:164]: Yes exactly, because we nee validators to advertise their ips

**Comment by eduardogr** [comment]
Closing this for new PR that replace this one: #928

**Pull request closed without merging**